### PR TITLE
ref(tests): DRY OrganizationMember tests.

### PR DESCRIPTION
**Pull request by mgaeta**
The files:
 - `test_organization_member_issues_assigned.py`
 - `test_organization_member_team_details.py`
 - `test_organization_member_unreleased_commits.py`
have a lot of repeated code.

This PR uses `get_success_response()` and moves repeated code to `setUp()`.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_member_team_details.py [hunk 1]
```diff
@@ -205,8 +192,7 @@ def org(self):
 
     def test_member_must_request_access_to_join_team(self):
         self.login_as(self.member.user)
-        resp = self.get_response(self.org.slug, self.member.id, self.team.slug)
-        assert resp.status_code == 202
+        self.get_success_response(self.org.slug, self.member.id, self.team.slug)
```
Comment (JoshFerge) [inline tests/sentry/api/endpoints/test_organization_member_team_details.py:195]: should we still look specifically for a 202 here? (and a 201 above)

Comment (mgaeta) [reply tests/sentry/api/endpoints/test_organization_member_team_details.py:195]: Yes this is the only file where we make assertions about the API so we should be explicit about status code if that's something we want to be strict about.

**Review by JoshFerge — COMMENTED** [review]
looks good! just the question about looking at status code since in that case I'd like us to be specific about it being a `202`.

File: tests/sentry/api/endpoints/test_organization_member_team_details.py [hunk 2]
```diff
@@ -258,12 +276,14 @@ def test_admin_must_request_access_to_add_member_to_team(self):
...
-        assert resp.status_code == 202
+        self.get_success_response(
+            self.org.slug, self.admin.id, self.team.slug, status=status.HTTP_201_CREATED
+        )
 
         self.login_as(self.team_member.user)
-        resp = self.get_response(self.org.slug, self.admin.id, self.team.slug)
-        assert resp.status_code == 202
+        self.get_success_response(
+            self.org.slug, self.admin.id, self.team.slug, status=status.HTTP_201_CREATED
```
Comment (JoshFerge) [inline tests/sentry/api/endpoints/test_organization_member_team_details.py:285]: shouldn't this be a 202?

Comment (JoshFerge) [reply tests/sentry/api/endpoints/test_organization_member_team_details.py:285]: how did this test pass? lol

**Review by JoshFerge — CHANGES_REQUESTED** [review]
want to make sure we assert correct status code before merging.

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**